hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueFactory.kt

```diff
@@ -58,60 +60,76 @@ class HmppsQueueFactory(
+    getOrDefaultBean("${hmppsQueue.id}-sqs-listener-factory") {
+      HmppsQueueDestinationContainerFactory(hmppsQueue.id, createSqsListenerContainerFactory(hmppsQueue.sqsClient))
+    }
+
+  fun createSqsListenerContainerFactory(awsSqsClient: SqsAsyncClient): SqsMessageListenerContainerFactory<String> =
+    SqsMessageListenerContainerFactory
+      .builder<String>()
+      .sqsAsyncClient(awsSqsClient)
+      .build()
+
+  fun createSqsAsyncDlqClient(queueConfig: QueueConfig, hmppsSqsProperties: HmppsSqsProperties): SqsAsyncClient {
     val region = hmppsSqsProperties.region
+    val provider = findProvider(hmppsSqsProperties.provider)
     if (queueConfig.dlqName.isEmpty()) throw MissingDlqNameException()
-    return when (findProvider(hmppsSqsProperties.provider)) {
-      Provider.AWS -> sqsClientFactory.awsSqsClient(queueConfig.dlqAccessKeyId, queueConfig.dlqSecretAccessKey, region)
-      Provider.LOCALSTACK -> sqsClientFactory.localstackSqsClient(hmppsSqsProperties.localstackUrl, region)
-        .also { sqsDlqClient -> sqsDlqClient.createQueue(CreateQueueRequest.builder().queueName(queueConfig.dlqName).build()) }
+    return when (provider) {
+      Provider.AWS -> sqsClientFactory.awsSqsAsyncClient(queueConfig.dlqAccessKeyId, queueConfig.dlqSecretAccessKey, region)
+      Provider.LOCALSTACK -> {
+        sqsClientFactory.localstackSqsAsyncClient(hmppsSqsProperties.localstackUrl, region)
+          .also { sqsDlqClient -> runBlocking { sqsDlqClient.createQueue(CreateQueueRequest.builder().queueName(queueConfig.dlqName).build()).await() } }
+      }
     }
   }
 
-  fun createSqsClient(queueConfig: QueueConfig, hmppsSqsProperties: HmppsSqsProperties, sqsDlqClient: SqsClient?): SqsClient {
+  fun createSqsAsyncClient(queueConfig: QueueConfig, hmppsSqsProperties: HmppsSqsProperties, sqsDlqClient: SqsAsyncClient?): SqsAsyncClient {
     val region = hmppsSqsProperties.region
     return when (findProvider(hmppsSqsProperties.provider)) {
-      Provider.AWS -> sqsClientFactory.awsSqsClient(queueConfig.queueAccessKeyId, queueConfig.queueSecretAccessKey, region)
-      Provider.LOCALSTACK -> sqsClientFactory.localstackSqsClient(hmppsSqsProperties.localstackUrl, region)
-        .also { sqsClient -> createLocalStackQueue(sqsClient, sqsDlqClient, queueConfig.queueName, queueConfig.dlqName, queueConfig.dlqMaxReceiveCount) }
+      Provider.AWS -> sqsClientFactory.awsSqsAsyncClient(queueConfig.queueAccessKeyId, queueConfig.queueSecretAccessKey, region)
+      Provider.LOCALSTACK -> {
+        sqsClientFactory.localstackSqsAsyncClient(hmppsSqsProperties.localstackUrl, region)
+          .also { sqsClient -> runBlocking { createLocalStackQueueAsync(sqsClient, sqsDlqClient, queueConfig.queueName, queueConfig.dlqName, queueConfig.dlqMaxReceiveCount) } }
+      }
     }
   }
 
-  private fun createLocalStackQueue(
-    sqsClient: SqsClient,
-    sqsDlqClient: SqsClient?,
+  private suspend fun createLocalStackQueueAsync(
```

Comment:
I didn't make this a suspend function previously because I was wondering what would happen if the request to create the queue fails for whatever reason. If it's synchronous everything falls over and the app fails to start...

Reply:
Or also if the queue is created but takes a long time, which could mean it's not there when the tests start running.